src/types/coordinate_nb.cpp

```diff
@@ -0,0 +1,81 @@
+#include "util/coordinate.hpp"
+
+#include <nanobind/nanobind.h>
+#include <nanobind/operators.h>
+#include <nanobind/stl/string.h>
+#include <nanobind/stl/pair.h>
+
+namespace nb = nanobind;
+using namespace nb::literals;
+
+void init_Coordinate(nb::module_& m) {    
+    namespace tag = osrm::util::tag;
+    using FixedLatitude = osrm::Alias<int32_t, tag::latitude>;
+    using FixedLongitude = osrm::Alias<int32_t, tag::longitude>;
+    using FloatLatitude = osrm::Alias<double, tag::latitude>;
+    using FloatLongitude = osrm::Alias<double, tag::longitude>;
+    using UnsafeFloatLatitude = osrm::Alias<double, tag::unsafelatitude>;
+    using UnsafeFloatLongitude = osrm::Alias<double, tag::unsafelongitude>;
+
+    using osrm::util::Coordinate;
+    using osrm::util::FloatCoordinate;
+
+    nb::class_<FixedLatitude>(m, "FixedLatitude")
+        .def(nb::init<>())
+        .def(nb::init<int32_t>());
+    nb::class_<FloatLatitude>(m, "FloatLatitude")
+        .def(nb::init<>())
+        .def(nb::init<double>());
+    nb::class_<UnsafeFloatLatitude>(m, "UnsafeFloatLatitude")
+        .def(nb::init<>())
+        .def(nb::init<double>())
+        .def(nb::init_implicit<double>());
+
+    nb::class_<FixedLongitude>(m, "FixedLongitude")
+        .def(nb::init<>())
+        .def(nb::init<int32_t>());
+    nb::class_<FloatLongitude>(m, "FloatLongitude")
+        .def(nb::init<>())
+        .def(nb::init<double>());
+    nb::class_<UnsafeFloatLongitude>(m, "UnsafeFloatLongitude")
+        .def(nb::init<>())
+        .def(nb::init<double>())
+        .def(nb::init_implicit<double>());
+
+    nb::class_<FloatCoordinate>(m, "FloatCoordinate")
+        .def(nb::init<>())
+        .def(nb::init<const Coordinate>(), "other"_a)
+        .def(nb::init<const FixedLongitude, const FixedLatitude>(),
+            "lon"_a, "lat"_a)
+        .def(nb::init<const FloatLongitude, const FloatLatitude>(),
+            "lon"_a, "lat"_a)
+        .def_rw("lon", &FloatCoordinate::lon)
+        .def_rw("lat", &FloatCoordinate::lat)
+        .def("IsValid", &FloatCoordinate::IsValid)
+        .def(nb::self == nb::self)
+        .def(nb::self != nb::self);
```

Comment:
what do these do?

Reply:
Those are bindings for operators - so in this case, it would allow `==` and `!=` comparisons between the Coordinate objects.